tests/e2e/fio_test.go

```diff
@@ -17,15 +17,80 @@ limitations under the License.
 package e2e
 
 import (
+	"fmt"
+	"io/ioutil"
+
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
+	"github.com/xridge/kubestone/api/v1alpha1"
+	batchv1 "k8s.io/api/batch/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	fioCrBaseDir    = samplesDir + "/fio"
+	e2eNamespaceFio = "kubestone-e2e-fio"
 )
 
 var _ = Describe("end to end test", func() {
-	Describe("for fio", func() {
-		// TODO: Add fio end-to-end test
-		It("Should be implemented", func() {
-			Expect(true).To(BeTrue())
+	Context("preparing namespace", func() {
+		_, _, err := run("kubectl create namespace " + e2eNamespaceFio)
+		It("should succeed", func() {
+			Expect(err).To(BeNil())
 		})
 	})
+
+	fioCrDirs := []string{fioCrBaseDir + "/base"}
+	fioOverlayContents, err := ioutil.ReadDir(fioCrBaseDir + "/overlays")
+	if err != nil {
+		Fail("Didn't find any fio CRs under " + fioCrBaseDir)
+	}
+	for _, fioOverlayContent := range fioOverlayContents {
+		if fioOverlayContent.IsDir() {
+			fioCrDirs = append(fioCrDirs, fioCrBaseDir+"/overlays/"+fioOverlayContent.Name())
+		}
+	}
+
+	Describe("creating fio job from multiple CRs", func() {
+		for i, fioCrDir := range fioCrDirs {
+			crName := fmt.Sprintf("fio-sample-%d", i)
```

Comment:
I just realized by looking at the run that this makes it impossible to figure out what went wrong:
we see fio-sample-0, fio-sample-1, etc. jobs, but the order is nondeterministic .

Can we please include here the crDir name, so that we could see what failed?

Reply:
done